### PR TITLE
As a user, my country location should be detected so I can browse and checkout faster because of prefilled values #59

### PR DESCRIPTION
**Pull request by AlexeiaW**
As a user, my country location should be detected so I can browse and checkout faster because of prefilled values #59 

Made a gelocation interface
Made a maxmind object that implements the gelocation interface
Made a Gelocation Service that interacts with the specific implementation of the geolocation interface, in our case the default binding is maxmind
Made a geolocation service that can be instantiated, and is injected with whatever geolocation implementation is provided.

### CODE REVIEW

File: src/CartServiceProvider.php [hunk 1]
```diff
@@ -6,7 +6,13 @@
 use GuzzleHttp\Client;
 use Illuminate\Support\ServiceProvider;
 use Lab19\Cart\Services\CartService;
+use Lab19\Cart\Services\CurrencyConversionInterface;
+use Lab19\Cart\Services\GeolocationInterface;
+use Lab19\Cart\Services\GeoLocationService;
```
Comment (luke-siedle) [inline src/CartServiceProvider.php]: @AlexeiaW let's make the GeoLocationService the same as the interface in terms of camelcase, so GeolocationService (no upper case L).

File: src/GraphQL/Mutations/SetSession.php [hunk 2]
```diff
@@ -40,4 +42,40 @@ public function setCurrency($rootValue, array $args, GraphQLContext $context, Re
 
         return $sessionService->get();
     }
+
+    // setGeoLocation
+    public function setGeoLocation($rootValue, array $args, GraphQLContext $context, ResolveInfo $resolveInfo)
+    {
+        $ip_address = $args['input']['ip_address'];
+
+        if (!isset($ip_address)) {
```
Comment (luke-siedle) [inline src/GraphQL/Mutations/SetSession.php]: @AlexeiaW for this part, we don't really want to give the user an opportunity to input an IP, rather we can just pull this value from $_SERVER['REMOTE_ADDR']

For testing we can use this phpunit feature to mock the IP address:

https://stackoverflow.com/a/53283002/558287

Technically speaking, we may want to run this method from inside the initial session token creation function, but this is a nice placeholder for this functionality for now. 

(This feature could be used to restrict certain shopping items to regions, so we wouldn't want the user to be able to set this manually in anyway, and rely on the IP reporting from their connection)

File: src/CartServiceProvider.php (hunk 1 above)
Comment (AlexeiaW) [reply src/CartServiceProvider.php]: Renamed the service

File: src/GraphQL/Mutations/SetSession.php (hunk 2 above)
Comment (AlexeiaW) [reply src/GraphQL/Mutations/SetSession.php]: Yeah it makes sense, we wouldn't want there be a way for external clients to change the ip address, or provide a fake one

**Pull request merged**